File: CursClient/src/app/entities/torrent/torrent.ts
import {Category} from "../category/category";
import {Permission} from "../permission/permission";
import {User} from "../user/user";
export class Torrent {
  public id: number;
  public name: string;
  public description: string;
  public user: User;
  public creationDate: string;
  public editDate: string;
  public status: string;
  public files?: File[];
  public comments?: Comment[];
  public category: Category;
  public permission: Permission;
  //


  constructor(id: number,
              name: string,
              description?: string,
              latestModificationDate?: string,
              createDate?: string,
              status?: string,
              files?: File[],
              comments?: Comment[]) {
    this.id = id;
    this.name = name;
    if (description)
      this.description = description;
    if (createDate)
      this.creationDate = createDate;
    if (latestModificationDate)
      this.editDate = latestModificationDate;
    if (status)
      this.status = status;
    if(files)
      this.files = files;
    if (comments)
      this.comments = comments;
  }
}

File: CursServer/src/main/java/core/torrent/model/Torrent.java
package core.torrent.model;

import core.category.model.Category;
import core.permission.model.Permission;
import core.user.model.User;

import javax.persistence.*;
import java.sql.Date;

@Entity
@Table(name = "torrents")
public class Torrent {
    @Id
    private int id;

    @ManyToOne()
    @JoinColumn(name = "category_id")
    private Category category;
    @ManyToOne()
    @JoinColumn(name = "user_id")
    private User user;
    @ManyToOne()
    @JoinColumn(name = "permission_id")
    private Permission permission;
    private String name;
    private String description;
    private Date creationDate;
    private Date editDate;
    private String status;

    public Torrent() {
    }

    public Category getCategory() {
        return category;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public Date getCreationDate() {
        return creationDate;
    }

    public Date getEditDate() {
        return editDate;
    }

    public String getStatus() {
        return status;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Permission getPermission() {
        return permission;
    }
}
